leptonai/cloudrun/remote.py

```diff
@@ -112,91 +123,125 @@ def start_up(self):
-
         # wait for the photon to be ready
         logger.debug(f"Remote: waiting for photon {self.photon_id} to be ready")
         start = time.time()
         is_deployment_ready = False
+        ret = None
         while not is_deployment_ready and time.time() - start < self._MAX_WAIT_TIME:
-            ret = api.deployment.get_deployment(conn, self.deployment_id)
+            ret = api.deployment.get_deployment(self.conn, self.deployment_id)
             if isinstance(ret, APIError):
                 raise RuntimeError(
                     f"Failed to get the status of deployment {self.deployment_id}."
                     f" Response: {ret.message}."
                 )
-            elif ret["status"]["state"] == "Running":  # noqa
+            elif ret["status"]["state"] in ("Running", "Ready"):  # noqa
+                # In earlier versions of the backend we had "Running" as a state.
+                # As a result we will temporarily use both, and in the long run we
+                # will use "Ready". ref: https://github.com/leptonai/lepton/pull/3152
                 is_deployment_ready = True
             else:
                 # Test if there are earlier terminations causing the deployment to fail
-                ret = api.deployment.get_termination(conn, self.deployment_id)
+                ret = api.deployment.get_termination(self.conn, self.deployment_id)
                 if not isinstance(ret, APIError) and len(ret):
                     # Earlier termination detected. Raise an error.
                     raise RuntimeError(
                         f"{self.deployment_id} seems to have failures. Inspect the"
                         f" failure using `lep deployment log -n {self.deployment_id}`."
+                        " After this, you can run cloudrun.clean() to cleanup old"
+                        " deployments and photons."
                     )
                 time.sleep(self._DEFAULT_WAIT_INTERVAL)
         if not is_deployment_ready:
             raise RuntimeError(
                 f"Failed to run photon {self.photon_id} in time. Last state: {ret}"
             )
 
+    def _create_client(self):
+        self.client = Client(current(), self.deployment_id)
+        start = time.time()
+        while not self.client.openapi and time.time() - start < self._MAX_WAIT_TIME:
+            # logger.debug("Remote: cannot get openapi, recreating client...")
+            time.sleep(self._DEFAULT_WAIT_INTERVAL)
+            self.client = Client(current(), self.deployment_id)
+        if not self.client.openapi:
+            raise RuntimeError(
+                f"Failed to get openapi for photon {self.photon_id} in time."
+            )
+        logger.debug(f"Remote: photon {self.deployment_id} is ready")
+
+    def _start_up(self) -> None:
+        """
+        Starts up the remote run.
+        """
+        self._create_and_push_photon()
+        self._run_photon()
+        # Now the photon is ready, we can create the client.
```

Comment:
should here call `self._create_client()`?

Reply:
Yep, fixed.